griptape/artifacts/boolean_artifact.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import annotations
+from typing import Any
+from attrs import define, field
+from griptape.artifacts import BaseArtifact
+
+
+@define
+class BooleanArtifact(BaseArtifact):
+    value: bool = field(converter=bool, metadata={"serializable": True})
```

Comment:
passing in a value on init will get coerced into a `bool`, whereas using `parse_bool` will check for the literal true or false. should that check be done on init instead?

Reply:
I think having these two methods of creating a `BooleanArtifact` seems reasonable. The default (init) way being a little more flexible, but still offering a stricter option for strings specifically.

If you don't offer the bool coercion option and the use case comes up, then what would that look like? like this: `BooleanArtifact(bool(x))`? I suppose that's easy enough, but `BooleanArtifact(x)` is a little better. Actually, the `parse_bool` wouldn't work for a literal `True` or `False`, so I definitely vote for keeping both creation options.

Reply:
the init method will always convert the input to a boolean. this can lead to suprising behavior sometimes where `BooleanArtifact("False").value is True`. `parse_bool` is to parse string literal values as a convenience. thinking about it now, seems more "pythonic" to keep the init as to whatever python interprets as a bool, and use `parse_bool` to parse string literal values as a convenience